controllers/awsmachine_controller.go

```diff
@@ -411,23 +411,26 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 }
 
 func (r *AWSMachineReconciler) deleteEncryptedBootstrapDataSecret(machineScope *scope.MachineScope, secretSvc services.SecretsManagerInterface) error {
-	// do nothing if there isn't asecret
-	if machineScope.GetSecretARN() == "" {
+	// do nothing if there isn't a secret
+	if machineScope.GetSecretPrefix() == "" {
 		return nil
 	}
```

Comment:
We probably should shortcut here if SecretCount == 0, and likely consider it an error condition.